vms/proposervm/post_fork_block_test.go

```diff
@@ -401,102 +428,114 @@ func TestBlockVerify_PostForkBlock_PChainHeightChecks(t *testing.T) {
 		switch {
 		case bytes.Equal(b, coreGenBlk.Bytes()):
 			return coreGenBlk, nil
-		case bytes.Equal(b, prntCoreBlk.Bytes()):
-			return prntCoreBlk, nil
+		case bytes.Equal(b, parentCoreBlk.Bytes()):
+			return parentCoreBlk, nil
 		default:
 			return nil, errUnknownBlock
 		}
 	}
 
-	prntProBlk, err := proVM.BuildBlock(context.Background())
+	parentBlk, err := proVM.BuildBlock(context.Background())
 	require.NoError(err)
 
-	require.NoError(prntProBlk.Verify(context.Background()))
-	require.NoError(proVM.SetPreference(context.Background(), prntProBlk.ID()))
+	require.NoError(parentBlk.Verify(context.Background()))
+	require.NoError(proVM.SetPreference(context.Background(), parentBlk.ID()))
+
+	// set VM to be ready to build next block. We set it to generate unsigned blocks
+	// for simplicity.
+	nextTime := parentBlk.Timestamp().Add(proposer.MaxVerifyDelay)
+	proVM.Set(nextTime)
 
-	prntBlkPChainHeight := pChainHeight
+	parentBlkPChainHeight := parentBlk.(*postForkBlock).PChainHeight()
 
 	childCoreBlk := &snowman.TestBlock{
 		TestDecidable: choices.TestDecidable{
 			IDV:     ids.Empty.Prefix(2222),
 			StatusV: choices.Processing,
 		},
-		ParentV:    prntCoreBlk.ID(),
-		BytesV:     []byte{2},
-		TimestampV: prntProBlk.Timestamp().Add(proposer.MaxVerifyDelay),
+		ParentV: parentCoreBlk.ID(),
+		BytesV:  []byte{2},
 	}
-
-	// child P-Chain height must not precede parent P-Chain height
-	childSlb, err := block.Build(
```

Comment:
once again a signed block and some unsigned blocks after. Not sure why. Moved to all unsigned blocks. To be revisited to extend UTs coverage over signed block.